packages/aws-cdk/lib/import.ts

```diff
@@ -0,0 +1,362 @@
+
+        ret.importResources.push(resource);
+        ret.resourceMap[resource.logicalId] = idProps;
+        delete contents[resource.logicalId];
+      } else {
+        print('%s: skipping', chalk.blue(descr));
+      }
+    }
+
+    const unknown = Object.keys(contents);
+    if (unknown.length > 0) {
+      warning(`Unrecognized resource identifiers in mapping file: ${unknown.join(', ')}`);
+    }
+
+    return ret;
+  }
+
+  /**
+   * Based on the provided resource mapping, prepare CFN structures for import (template,
+   * ResourcesToImport structure) and perform the import operation (CloudFormation deployment)
+   *
+   * @param resourceMap Mapping from CDK construct tree path to physical resource import identifiers
+   * @param options Options to pass to CloudFormation deploy operation
+   */
+  public async importResources(importMap: ImportMap, options: DeployStackOptions) {
+    const resourcesToImport: ResourcesToImport = await this.makeResourcesToImport(importMap);
+    const updatedTemplate = await this.currentTemplateWithAdditions(importMap.importResources);
+
+    try {
+      const result = await this.cfn.deployStack({
+        ...options,
+        overrideTemplate: updatedTemplate,
+        resourcesToImport,
+      });
+
+      const message = result.noOp
+        ? ' ✅  %s (no changes)'
+        : ' ✅  %s';
+
+      success('\n' + message, options.stack.displayName);
+    } catch (e) {
+      error('\n ❌  %s failed: %s', chalk.bold(options.stack.displayName), e);
+      throw e;
+    }
+  }
+
+  /**
+   * Perform a diff between the currently running and the new template, enusre that it is valid
+   * for importing and return a list of resources that are being added in the new version
+   *
+   * @return mapping logicalResourceId -> resourceDifference
+   */
+  public async discoverImportableResources(allowNonAdditions = false): Promise<Array<ImportableResource>> {
+    const currentTemplate = await this.currentTemplate();
+
+    const diff = cfnDiff.diffTemplate(currentTemplate, this.stack.template);
+
+    // Ignore changes to CDKMetadata
+    const resourceChanges = Object.entries(diff.resources.changes)
+      .filter(([logicalId, _]) => logicalId !== 'CDKMetadata');
+
+    // Split the changes into additions and non-additions. Imports only make sense
+    // for newly-added resources.
+    const nonAdditions = resourceChanges.filter(([_, dif]) => !dif.isAddition);
+    const additions = resourceChanges.filter(([_, dif]) => dif.isAddition);
+
+    if (nonAdditions.length && !allowNonAdditions) {
+      const offendingResources = nonAdditions.map(([logId, _]) => this.describeResource(logId));
+      throw new Error('No resource updates or deletes are allowed on import operation. Make sure to resolve pending changes ' +
+                      `to existing resources, before attempting an import. Updated/deleted resources: ${offendingResources.join(', ')}`);
+    }
```

Comment:
Would want to print a warning in case those are allowed, so the user is aware that some changes were detected that will not be applied with the import.